drivers/adc/adc_nxp_s32_adc_sar.c

```diff
@@ -403,6 +403,18 @@ static void adc_nxp_s32_isr(const struct device *dev)
 #define ADC_NXP_S32_GET_INSTANCE(n)		\
 	LISTIFY(__DEBRACKET ADC_INSTANCE_COUNT, ADC_NXP_S32_INSTANCE_CHECK, (|), n)
 
+#if (FEATURE_ADC_HAS_HIGH_SPEED_ENABLE == 1U)
+#define ADC_NXP_S32_HIGH_SPEED_CFG(n) (.HighSpeedConvEn = DT_INST_PROP(n, high_speed),)
```

Comment:
my mental compilation may fail here, but this will expand to something like `struct xx = { (.HighSpeedConvEn = xxxx ,) }` that will fail to build because of the `( )`?

Also consider using `IF_ENABLED()` macro directly into the struct definition where this is being used.

Reply:
I removed the `()` from define.
About using `IF_ENABLE()`, It can't be used because the `FEATURE_ADC_HAS_HIGH_SPEED_ENABLE` has been defined as `(0U)/(1U)`.